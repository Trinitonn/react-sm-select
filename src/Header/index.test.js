import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Header } from './index';

describe('Header', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <Header
        disabled={false}
        expanded={true}
        focused={true}
        selected={true}
      >
        <div>Some Children</div>
      </Header>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly if disabled and not focused', () => {
    const wrapper = shallow(
      <Header
        disabled={true}
        expanded={false}
        focused={false}
        selected={false}
      >
        <div>Some Children</div>
      </Header>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly if disabled and focused', () => {
    const wrapper = shallow(
      <Header
        disabled={true}
        expanded={false}
        focused={true}
        selected={false}
      >
        <div>Some Children</div>
      </Header>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly if enabled, selected and focused', () => {
    const wrapper = shallow(
      <Header
        disabled={false}
        expanded={false}
        focused={true}
        selected={true}
      >
        <div>Some Children</div>
      </Header>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
