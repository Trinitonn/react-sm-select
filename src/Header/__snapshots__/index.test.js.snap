// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
<div
  aria-disabled={false}
  aria-expanded={true}
  aria-readonly="true"
  className="Header Header--focusedActive Header--expanded Header--selected"
  role="combobox"
  tabIndex="0"
>
  <div>
    Some Children
  </div>
</div>
`;

exports[`Header should render properly if disabled and focused 1`] = `
<div
  aria-disabled={true}
  aria-expanded={false}
  aria-readonly="true"
  className="Header Header--disabled Header--focusedDisabled"
  role="combobox"
  tabIndex="0"
>
  <div>
    Some Children
  </div>
</div>
`;

exports[`Header should render properly if disabled and not focused 1`] = `
<div
  aria-disabled={true}
  aria-expanded={false}
  aria-readonly="true"
  className="Header Header--disabled"
  role="combobox"
  tabIndex="0"
>
  <div>
    Some Children
  </div>
</div>
`;

exports[`Header should render properly if enabled, selected and focused 1`] = `
<div
  aria-disabled={false}
  aria-expanded={false}
  aria-readonly="true"
  className="Header Header--focusedActive Header--selected"
  role="combobox"
  tabIndex="0"
>
  <div>
    Some Children
  </div>
</div>
`;
